### PR TITLE
authority_client: rewrite udp address to tcp

### PR DESCRIPTION
**Pull request by bmwill**
## Description 

Describe the changes or additions included in this PR.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Review by pei-mysten — APPROVED** [review]

**Review by lxfind — APPROVED** [review]

File: crates/sui-core/src/authority_client.rs [hunk 1]
```diff
@@ -260,9 +260,15 @@ pub fn make_network_authority_clients_with_network_config(
...
-            .connect_lazy(address)
-            .map_err(|err| anyhow!(err.to_string()))?;
+        let address = address.rewrite_udp_to_tcp();
+        let channel = network_config.connect_lazy(&address).map_err(|e| {
+            tracing::error!(
+                address = %address,
+                name = %name,
+                "unable to create authority client: {e}"
+            );
+            anyhow!(err.to_string())
```
Comment (jin-eg) [inline crates/sui-core/src/authority_client.rs:270]: `anyhow!(e.to_string())` ?

Comment (wakiyamap) [inline crates/sui-core/src/authority_client.rs:270]: `anyhow!(e.to_string())` ?

Comment (bmwill) [reply crates/sui-core/src/authority_client.rs]: yep, i was too quick on the draw :)

Comment (wakiyamap) [reply crates/sui-core/src/authority_client.rs]: I cheer for Sui team!

**Pull request merged**